src/Faucet.sol

```diff
@@ -43,11 +73,8 @@ contract Faucet is Ownable, IFaucet {
      * @dev Claim via meta tx on behalf of a new account by the owner
      * @param _signatureData Signature data
      */
-    function claimOnBehalf(IFaucet.SignatureData calldata _signatureData)
-        external
-        onlyOwner
-        onlySignerVerified(_signatureData)
-    {
+    function claimOnBehalf(IFaucet.SignatureData calldata _signatureData) external onlySignerVerified(_signatureData) {
+        require(msg.sender == address(walletFactory), "Only wallet factory can call this");
```

Comment:
why do we have this requirement here but claimKintoETH does not?

Reply:
maybe for wallets late to claim directly